jax/custom_derivatives.py

```diff
@@ -818,7 +818,10 @@ def bwd(res, cts):
     cts_flat, out_tree_ = tree_flatten((cts,))
     if out_tree != out_tree_: raise TypeError(f'{out_tree}\n!=\n{out_tree_}')
     cts_out = core.eval_jaxpr(jaxpr, consts, *cts_flat)
-    return tree_unflatten(in_tree, cts_out)
+    cts_out = tree_unflatten(in_tree, cts_out)
+    if not isinstance(cts_out, tuple):
```

Comment:
```suggestion
    if tree_util.treedef_is_leaf(cts_out)
```

I'm thinking we could check whether the user returned a singleton, rather than just checking that it's not a tuple. WDYT?

Reply:
Good, idea, done.